sympy/geometry/tests/test_line.py

```diff
@@ -76,6 +76,12 @@ def test_arbitrary_point():
     assert Segment3D(Point3D(x1, x1, x1), Point3D(y1, y1, y1)).length == sqrt(3) * sqrt((x1 - y1) ** 2)
     assert Segment3D((1, 1, 1), (2, 3, 4)).arbitrary_point() == \
            Point3D(t + 1, 2 * t + 1, 3 * t + 1)
+    try:
+        Line((x,1),(2,3)).arbitrary_point(x)
+    except ValueError:
+        assert True
+    except BaseException:
+        assert False
```

Comment:
You can use `raises` from `sympy.utilities.pytest` to test correct error is raised. See `test_raises` around line 649.

Reply:
[BaseException](https://docs.python.org/2/library/exceptions.html#exceptions.BaseException) is only used as a base class for other exceptions.

Reply:
@ylemkimon The code as written is progmatically correct, the test is passed if a ValueError is thrown and fails if any other error is thrown. BaseException is the superclass of all exceptions in python therefore when an exception is inevitably thrown, either it will be a ValueError and will be caught by the first except clause, or it will be caught by the second except clause. This functions as intended.

Reply:
@jksuom is telling you how we test such things in the test suite. The `raises` function will do what you are saying with less noise.

Reply:
I see what you mean now, what do you mean by noise though?